src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6283,12 +6292,13 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
                         {
                             lastTokenOfType = this.EatToken();
                         }
-                    } while (this.CurrentToken.Kind == SyntaxKind.CommaToken);
+                    }
+                    while (this.CurrentToken.Kind == SyntaxKind.CommaToken);
 
                     if (this.CurrentToken.Kind == SyntaxKind.CloseParenToken)
                     {
                         lastTokenOfType = this.EatToken();
-                        return ScanTypeFlags.MustBeType;
+                        return ScanTypeFlags.TupleType;
```

Comment:
@gafter I changed this so that is would return `MustBeType` if any of the element types are also deemed `MustBeType`.